src/vsr/replica.zig

```diff
@@ -5276,6 +5289,8 @@ pub fn ReplicaType(
             var iterator = self.pipeline.iterator();
             while (iterator.next_ptr()) |prepare| {
                 assert(prepare.message.header.command == .prepare);
+                assert(!prepare.ok_quorum_received);
```

Comment:
Just to double-check, are we vulnerable to an assertion crash here, for example, if a `prepare_ok` arrives in between resetting the quorum and before calling `verify_pipeline()`? Can this happen?

Reply:
No, that can't happen because:

- The replica resets the quorum when transitioning to `normal` or `view_change` status.
- `ignore_prepare_ok` rejects `prepare_ok`s when the replica is in `view_change`, so we only need to worry about `normal`.
- But `verify_pipeline` is called from:
  - `start_view_as_the_new_leader` (_before_ setting `status = normal`, so the `status` is still `view_change`).
  - And `repair_pipeline_diff`, which is only called during `status == view_change`.

So not only is it safe, but I can add `assert(self.status == .view_change)` to `verify_pipeline`.

Reply:
Nice!